tapo/src/api/child_devices/ke100_handler.rs

```diff
@@ -0,0 +1,176 @@
+            .await?;
+
+        Ok(())
+    }
+
+    /// Sets the *maximum control temperature*.
+    ///
+    /// # Arguments
+    ///
+    /// * `max_control_temperature`
+    pub async fn set_max_control_temperature(&self, max_control_temperature: u8) -> Result<(), Error> {
+        let json = serde_json::to_value(TrvSetDeviceInfoParams::new().max_control_temp(max_control_temperature)?)?;
+        let request = TapoRequest::SetDeviceInfo(Box::new(TapoParams::new(json)));
+
+        self.hub_handler
+            .control_child::<serde_json::Value>(self.device_id.clone(), request)
+            .await?;
+
+        Ok(())
+    }
+
+    /// Sets frost protection on the device to *on* or *off*.
+    ///     
+    /// # Arguments
+    ///
+    /// * `frost_protection_on` - true/false
+    pub async fn set_frost_protection(&self, frost_protection_on: bool) -> Result<(), Error> {
+        let json = serde_json::to_value(TrvSetDeviceInfoParams::new().frost_protection_on(frost_protection_on)?)?;
+        let request = TapoRequest::SetDeviceInfo(Box::new(TapoParams::new(json)));
+    
+        self.hub_handler
+            .control_child::<serde_json::Value>(self.device_id.clone(), request)
+            .await?;
+
+        Ok(())
+    }
+
+    /// Sets child protection on the device to *on* or *off*.
+    ///     
+    /// # Arguments
+    ///
+    /// * `child_protection_on` - true/false
+    pub async fn set_child_protection(&self, child_protection_on: bool) -> Result<(), Error> {
+        let json = serde_json::to_value(TrvSetDeviceInfoParams::new().child_protection(child_protection_on)?)?;
+        let request = TapoRequest::SetDeviceInfo(Box::new(TapoParams::new(json)));
+    
+        self.hub_handler
+            .control_child::<serde_json::Value>(self.device_id.clone(), request)
+            .await?;
+
+        Ok(())
+    }
+
+    /// Sets the *temperature offset*.
+    ///
+    /// # Arguments
+    ///
+    /// * `temp_offset` - between 5 and 30
+    pub async fn set_temp_offset(&self, temp_offset: i8) -> Result<(), Error> {
+        let json = serde_json::to_value(TrvSetDeviceInfoParams::new().temp_offset(temp_offset)?)?;
+        let request = TapoRequest::SetDeviceInfo(Box::new(TapoParams::new(json)));
+
+        self.hub_handler
+            .control_child::<serde_json::Value>(self.device_id.clone(), request)
+            .await?;
+
+        Ok(())
+    }
+
+    /// Returns *min_control_temp* and *max_control_temp* as Vec<u8>.
+    async fn get_control_range(&self) -> Result<Vec<u8>, Error> {
+            let request = TapoRequest::GetDeviceInfo(TapoParams::new(EmptyParams));
+    
+            self.hub_handler
+                .control_child::<KE100Result>(self.device_id.clone(), request)
+                .await?
+                .ok_or_else(|| Error::Tapo(TapoResponseError::EmptyResult))
+                .map(|result| vec![result.min_control_temp,result.max_control_temp])
+        }
```

Comment:
I need `get_control_range()` in `set_temperature()` to check if `target_temperature` is in the allowed range and present a proper error message with the correct range. Otherwise the device just sends an error code, which doesn't give an indication that `target_temperature` is out of the allowed range. I thus suggest to keep the function.

Reply:
Yes, but it would be simpler to call the existing `get_device_info` function inside `set_temperature()` and use `device_info. min_control_temp` and `device_info. max_control_temp` in the `if` clause directly.

Reply:
Yes, makes total sense to use device_info directly. I'll update the code.